code/model/Subsite.php

```diff
@@ -750,9 +758,41 @@ public function validate()
     public function onAfterWrite()
     {
         Subsite::writeHostMap();
+        $this->createDefaultPages();
```

Comment:
I think this can be done via `if ($this->isChanged('ID')) {` which would avoid the need for the new member property and the abstraction to the new function (although a protected function isn't really the worst thing). If this remains true (I'm 90% certain this was the case in SS3 at least) then it would make the code more readable and easier to follow.

Reply:
It would, but I wouldn't be at all surprised if this doesn't work. The last time I tried to track field level changes during a write I had to do this individually for every field I wanted to track because the framework calls ->write() multiple times. Nevermind, let's change the API and see if the test still passes!

Reply:
Works like a charm! Thanks for the suggestion 👍